testv2/e2e/helpers.go

```diff
@@ -86,22 +94,111 @@ func requiredTemplateFunc(warn string, input interface{}) (interface{}, error) {
 	return input, nil
 }
 
-func newKubeoneBin(terraformPath, manifestPath string) *kubeoneBin {
-	return &kubeoneBin{
-		bin:          "kubeone",
+func makeBin(args ...string) *testutil.Exec {
+	return testutil.NewExec("make",
+		testutil.WithArgs(args...),
+		testutil.WithEnv(os.Environ()),
+		testutil.InDir(filepath.Clean("../../")),
+		testutil.StdoutDebug,
+	)
+}
+
+func downloadKubeone(t *testing.T, version string) string {
+	binPath := filepath.Join(t.TempDir(), fmt.Sprintf("kubeone-%s", version))
+	zipPath := fmt.Sprintf("%s.zip", binPath)
+
+	exists, err := k8spath.Exists(k8spath.CheckSymlinkOnly, binPath)
+	if err != nil {
+		t.Fatalf("checking if kubeone already downloaded: %v", err)
+	}
+
+	if exists {
+		return binPath
+	}
+
+	const urlTemplate = "https://github.com/kubermatic/kubeone/releases/download/v%s/kubeone_%s_linux_amd64.zip"
+	downloadURL := fmt.Sprintf(urlTemplate, version, version)
+
+	req, err := http.NewRequest("GET", downloadURL, nil)
+	if err != nil {
+		t.Fatalf("building http request to download kubeone: %v", err)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		t.Fatalf("http request to download kubeone: %v", err)
+	}
+	defer resp.Body.Close()
+
+	zipBin, err := os.OpenFile(zipPath, os.O_CREATE|os.O_WRONLY, 0600)
+	if err != nil {
+		t.Fatalf("open kubeone destination file: %v", err)
+	}
+	defer zipBin.Close()
+
+	_, err = io.Copy(zipBin, resp.Body)
+	if err != nil {
+		t.Fatalf("downloading kubeone: %v", err)
+	}
+
+	fi, err := zipBin.Stat()
+	if err != nil {
+		t.Fatalf("file stat: %v", err)
+	}
+
+	unzip, err := zip.NewReader(zipBin, fi.Size())
+	if err != nil {
+		t.Fatalf("opening zip file for reading: %v", err)
+	}
+
+	unzipK1Bin, err := unzip.Open("kubeone")
+	if err != nil {
+		t.Fatalf("opening kubeone file from zip archive: %v", err)
+	}
+	defer unzipK1Bin.Close()
+
+	k1Bin, err := os.OpenFile(binPath, os.O_CREATE|os.O_WRONLY, 0750)
+	if err != nil {
+		t.Fatalf("open kubeone destination file: %v", err)
+	}
+	defer k1Bin.Close()
+
+	_, err = io.Copy(k1Bin, unzipK1Bin)
+	if err != nil {
+		t.Fatalf("extractive kubeone from zip: %v", err)
```

Comment:
```suggestion
		t.Fatalf("extracting kubeone from zip: %v", err)
```

`extracting or copying`

Reply:
Fixed :+1:

Reply:
Nice catch, even spell checker been confused :D